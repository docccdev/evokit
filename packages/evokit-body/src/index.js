import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
