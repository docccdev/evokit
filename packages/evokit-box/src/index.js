// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
