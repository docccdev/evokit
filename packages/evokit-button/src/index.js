import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
