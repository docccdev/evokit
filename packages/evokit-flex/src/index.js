import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
