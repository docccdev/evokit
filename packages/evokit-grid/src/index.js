import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
