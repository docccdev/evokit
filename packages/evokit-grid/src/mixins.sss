@define-mixin media $selector
    $selector
        @mixin-content

    @media (--ek-grid-media-small)
        $(selector)\@small
            @mixin-content

    @media (--ek-grid-media-medium)
        $(selector)\@medium
            @mixin-content

    @media (--ek-grid-media-large)
        $(selector)\@large
            @mixin-content

    @media (--ek-grid-media-wide)
        $(selector)\@wide
            @mixin-content

    @media (--ek-grid-media-huge)
        $(selector)\@huge
            @mixin-content


@define-mixin grid-divider-item $value
    &.grid__item
        & + .grid__item
            border-left-width: 0

        @for $index from 1 to $value
            &:nth-child(resolve($value)n-resolve($value - $index))
                @if $index == $value
                    border-right-width: 0
                @else
                    border-right-width: 1px

                &:nth-last-child(-n+resolve($value - $index + 1))
                    border-bottom-width: 0

                &:not(:nth-last-child(-n+resolve($value - $index + 1)))
                    border-bottom-width: 1px

@define-mixin grid-divider-item-expand
    &.grid__item
        & + .grid__item
            border-left-width: 1px

        &:nth-child(even),
        &:nth-child(odd)
            border-right-width: 0

            &:not(:nth-child(odd)),
            &:not(:nth-child(even))
                border-bottom-width: 0
