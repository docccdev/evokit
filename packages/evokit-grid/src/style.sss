@import './mixins.sss'

@custom-media --ek-grid-media-small only screen and (min-width: 480px)
@custom-media --ek-grid-media-medium only screen and (min-width: 768px)
@custom-media --ek-grid-media-large only screen and (min-width: 960px)
@custom-media --ek-grid-media-wide only screen and (min-width: 1200px)
@custom-media --ek-grid-media-huge only screen and (min-width: 1400px)

:root
    // prop 'grid-indent'
    --ek-grid-indent-xxs: 5px
    --ek-grid-indent-xs: 10px
    --ek-grid-indent-s: 15px
    --ek-grid-indent-m: 20px
    --ek-grid-indent-l: 25px
    --ek-grid-indent-xl: 30px
    --ek-grid-indent-xxl: 35px

.grid
    display: flex
    flex-wrap: wrap
    min-width: 0

.grid__item
    box-sizing: border-box
    min-width: 0
    max-width: 100%

/**
 * Indent:
 *
 *  - grid_indent_*
 *  - grid_indent-x_*
 *  - grid_indent-y_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .grid_indent_$(value)
        @if $value == none
            margin: 0

            & > .grid__item
                padding: 0

            &[class*='grid_divider']
                margin: 0

                & > .grid__item
                    padding: 0
        @else
            margin: calc(var(--ek-grid-indent-$(value)) / 2 * -1)

            & > .grid__item
                padding: calc(var(--ek-grid-indent-$(value)) / 2)

            &[class*='grid_divider']
                margin: 0

                & > .grid__item
                    padding: var(--ek-grid-indent-$(value))

    @mixin media .grid_indent-x_$(value)
        @if $value == none
            margin-left: 0
            margin-right: 0

            & > .grid__item
                padding-left: 0
                padding-right: 0

            &[class*='grid_divider']
                margin-left: 0
                margin-right: 0

                & > .grid__item
                    padding-left: 0
                    padding-right: 0
        @else
            margin-left: calc(var(--ek-grid-indent-$(value)) / 2 * -1)
            margin-right: calc(var(--ek-grid-indent-$(value)) / 2 * -1)

            & > .grid__item
                padding-left: calc(var(--ek-grid-indent-$(value)) / 2)
                padding-right: calc(var(--ek-grid-indent-$(value)) / 2)

            &[class*='grid_divider']
                margin-left: 0
                margin-right: 0

                & > .grid__item
                    padding-left: var(--ek-grid-indent-$(value))
                    padding-right: var(--ek-grid-indent-$(value))

    @mixin media .grid_indent-y_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0

            & > .grid__item
                padding-top: 0
                padding-bottom: 0

            &[class*='grid_divider']
                margin-top: 0
                margin-bottom: 0

                & > .grid__item
                    padding-top: 0
                    padding-bottom: 0
        @else
            margin-top: calc(var(--ek-grid-indent-$(value)) / 2 * -1)
            margin-bottom: calc(var(--ek-grid-indent-$(value)) / 2 * -1)

            & > .grid__item
                padding-top: calc(var(--ek-grid-indent-$(value)) / 2)
                padding-bottom: calc(var(--ek-grid-indent-$(value)) / 2)

            &[class*='grid_divider']
                margin-top: 0
                margin-bottom: 0

                & > .grid__item
                    padding-top: var(--ek-grid-indent-$(value))
                    padding-bottom: var(--ek-grid-indent-$(value))

/**
 * Colum:
 *
 *  - grid_column_*
 *
 **/

@each $value in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @mixin media .grid_column_$(value)
        & > .grid__item
            flex: none
            width: calc(100% / $value)

        &[class*='grid_divider']

            & > .grid__item
                @mixin grid-divider-item $value

@mixin media .grid_column_expand
    & > .grid__item
        flex: 1
        width: auto

@mixin media .grid_column_auto
    & > .grid__item
        flex: none
        width: auto

/**
 * Align:
 *
 *  - grid_align_*
 *
 **/

@each $value in (left, center, right, justify)
    @mixin media .grid_align_$(value)
        @if $value == left
            justify-content: flex-start
        @else if $value == center
            justify-content: center
        @else if $value == right
            justify-content: flex-end
        @else if $value == justify
            justify-content: space-between

/**
 * Valign:
 *
 *  - grid_valign_*
 *
 **/

@each $value in (top, middle, bottom, baseline, justify)
    @mixin media .grid_valign_$(value)
        @if $value == top
            align-items: flex-start
        @else if $value == middle
            align-items: center
        @else if $value == bottom
            align-content: flex-end
            align-items: flex-end
        @else if $value == baseline
            align-items: baseline
        @else if $value == justify
            align-content: space-between

/**
 * Direction:
 *
 *  - grid_direction_*
 *
 **/

@each $value in (row, row-reverse, column, column-reverse)
    @mixin media .grid_direction_$(value)
        flex-direction: $value

/**
 * Wrap:
 *
 *  - grid_wrap_*
 *
 **/

@each $value in (nowrap, wrap, wrap-reverse)
    @mixin media .grid_wrap_$(value)
        flex-wrap: $value

/**
 * Item order:
 *
 *  - grid__item_order_*
 *
 **/

@each $value in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @mixin media .grid__item_order_$(value)
        order: $value

/**
 * Width:
 *
 *  - grid__item_width_*
 *
 **/

@each $to in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @each $from in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
        @if $from < $to or ($from == 1 and $to == 1)
            @mixin media .grid__item_width_$(from)-$(to)
                flex: none
                width: calc($from / $to * 100%)

@mixin media .grid__item_width_expand
    flex: 1
    width: auto

@mixin media .grid__item_width_auto
    flex: none
    width: auto
