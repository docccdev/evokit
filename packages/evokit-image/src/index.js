import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
