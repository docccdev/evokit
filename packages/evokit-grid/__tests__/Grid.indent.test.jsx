import React from 'react';
import { shallow } from 'enzyme';
import { Grid } from '../src';

describe('<Grid grid-indent="..." />', () => {
    it('props grid-indent', () => {
        const wrapper = shallow(<Grid grid-indent='xs' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-grid ek-grid_indent_xs"></div>');
    });

    it('extend 2 props grid-indent', () => {
        const wrapper = shallow(<Grid grid-indent='xs xl' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-grid ek-grid_indent-x_xs ek-grid_indent-y_xl"></div>');
    });

    it('props grid-indent-x', () => {
        const wrapper = shallow(<Grid grid-indent-x='xs' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-grid ek-grid_indent-x_xs"></div>');
    });

    it('props grid-indent-y', () => {
        const wrapper = shallow(<Grid grid-indent-y='xs' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-grid ek-grid_indent-y_xs"></div>');
    });
});
