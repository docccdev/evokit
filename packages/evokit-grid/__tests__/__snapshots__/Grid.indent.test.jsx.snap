// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid grid-indent="..." /> extend 2 props grid-indent 1`] = `
<div
  className="ek-grid ek-grid_indent-x_xs ek-grid_indent-y_xl"
/>
`;

exports[`<Grid grid-indent="..." /> props grid-indent 1`] = `
<div
  className="ek-grid ek-grid_indent_xs"
/>
`;

exports[`<Grid grid-indent="..." /> props grid-indent-x 1`] = `
<div
  className="ek-grid ek-grid_indent-x_xs"
/>
`;

exports[`<Grid grid-indent="..." /> props grid-indent-y 1`] = `
<div
  className="ek-grid ek-grid_indent-y_xs"
/>
`;
